File: app/src/main/java/com/example/blackjackformoreplayersmobileapp/classes/Player.java
package com.example.blackjackformoreplayersmobileapp.classes;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public class Player implements Comparable<Player> {
    private int sumaValue = 0;
    final private int id;
    private String name;
    private final ArrayList<Card> cardCollection = new ArrayList<>();

    public Player(String name, int id) {
        this.name = name;
        this.id = id;
    }
    public int getSumaValueCards() {
        return this.sumaValue;
    }
    public String getName() {
        return this.name;
    }
    public int getId() {
        return this.id;
    }
    public ArrayList<Card> getCardCollection() {
        return this.cardCollection;
    }
    public void setSumaValue(int sumaValue) {
        this.sumaValue = sumaValue;
    }
    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Player" + this.id + " : " + this.name + " with cards: " + this.cardCollection + " suma of cards is " + this.sumaValue;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Player that = (Player) o;
        return sumaValue == that.sumaValue && id == that.id && Objects.equals(name, that.name) && Objects.equals(cardCollection, that.cardCollection);
    }
    @Override
    public int compareTo(Player anotherPlayer) {
        return this.getSumaValueCards() - anotherPlayer.getSumaValueCards();
    }

    public void takeCard(List<Card> cardsList) {
        int randomIndex = (int) (Math.random() * cardsList.size());
        Card card = cardsList.get(randomIndex);
        this.cardCollection.add(card);
        this.sumaValue += card.getCardValue().getVALUE();
        cardsList.remove(randomIndex);
    }

}

File: app/src/main/java/com/example/blackjackformoreplayersmobileapp/model/Player.kt
package com.example.blackjackformoreplayersmobileapp.model


class Player(var name: String, val id: Int) : Comparable<Player> {
    var sumValueCards = 0
    val cardCollection = mutableListOf<Card>()


    override fun toString(): String {
        return "Player$id : $name with cards: $cardCollection suma of cards is $sumValueCards"
    }

    override fun compareTo(other: Player): Int {
        return this.sumValueCards - other.sumValueCards
    }

    fun takeCard(cardsList: MutableList<Card>) {
        val randomIndex = (Math.random() * cardsList.size).toInt()
        val card = cardsList[randomIndex]
        cardCollection.add(card)
        sumValueCards += card.cardValue.value
        cardsList.elementAt(randomIndex)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Player

        if (name != other.name) return false
        if (id != other.id) return false
        if (sumValueCards != other.sumValueCards) return false
        if (cardCollection != other.cardCollection) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + id
        result = 31 * result + sumValueCards
        result = 31 * result + cardCollection.hashCode()
        return result
    }
}